File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.js
/**
 * @param {number} n
 * @return {number}
 */
var countCommas = function(n) {
     
        let cur = 1000;
        let res = 0n;
        
        // Convert to BigInt to handle large numbers safely in JavaScript
        let bigN = BigInt(n);
        let bigCur = BigInt(cur);

        while (bigCur <= bigN) {
            res += bigN - bigCur + 1n;
            bigCur *= 1000n;
        }   
        return Number(res);
}

File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.c
long long countCommas(long long n) {
    long long cur = 1000;
    long long res = 0;

    while (cur <= n) {
        res += n - cur + 1;
        // Check for potential overflow before multiplying by 1000
        if (cur > LLONG_MAX / 1000) {
            break;
        }
        cur *= 1000;
    }   
    return res;
}